index.ts

```diff
@@ -82,8 +82,10 @@ async function main() {
     if (!manifestKey && !buildCacheKey) {
       await run("Run esy cleanup", "esy", ["cleanup", "."]);
     }
-  } catch (e) {
-    core.setFailed(e.message);
+  } catch (error) {
+    if (error instanceof Error) {
+      core.error(error.message);
+    }
```

Comment:
This is something I've been doing recently.
```suggestion
    } else {
      core.error(util.inspect(error))
    }
```